libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -382,8 +382,8 @@ object MapboxRouteLineUtils {
     }
 
     private fun getRoadClassForIndex(roadClassArray: Array<String?>, index: Int): String? {
```

Comment:
If this is changed to:
private tailrec fun getRoadClassForIndex...
The kotlin compiler will optimize out the recursion.

Reply:
Good find! Let's follow up :+1: